### PR TITLE
fix(ci): make sure Rust tests actually run in deploy-gcp-tests.yml

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

It's easy to accidentally skip a test in CI, and not notice.
We want to check that we're actually running tests, and that they are succeeding.

Part of #2653.

Depends-On: #4704

## Solution

In `deploy-gcp-tests.yml`:
- before checking the exit status, check for "test result: 1 passed" in the logs
- use a regex to allow multiple tests, even though we only currently run one
- fail if the test logs don't show a success, or the test command returns a failing exit status

This PR does not check tests in any other workflows.

## Review

This is a low priority, but it seemed easy to fix while I was in that file.

### Reviewer Checklist

  - [ ] Existing tests pass

### CODE REVIEW

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
@mergifyio update

File: .github/workflows/deploy-gcp-tests.yml [hunk 1]
```diff
@@ -834,7 +834,9 @@ jobs:
           service_account: 'github-service-account@zealous-zebra.iam.gserviceaccount.com'
           token_format: 'access_token'
 
-      # Wait for the container to finish, then exit with the test's exit status.
+      # Check that the container executed at least 1 Rust test harness test, and that all tests passed.
```
Comment (dconnolly) [inline .github/workflows/deploy-gcp-tests.yml:837]: 👍

**Review by dconnolly — APPROVED** [review]

**Comment by teor2345** [comment]
@mergifyio refresh

**Pull request merged**